src/main/java/cubicchunks/worldgen/gui/component/UIItemGrid.java

```diff
@@ -0,0 +1,83 @@
+/*
+ *  This file is part of Cubic Chunks Mod, licensed under the MIT License (MIT).
+ *
+ *  Copyright (c) 2015 contributors
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package cubicchunks.worldgen.gui.component;
+
+import cubicchunks.worldgen.gui.ExtraGui;
+import net.malisis.core.client.gui.component.UIComponent;
+import net.minecraft.item.ItemBlock;
+import net.minecraft.item.ItemStack;
+
+public class UIItemGrid extends UILayout<UIItemGrid, Integer> {
+
+    UIFlatTerrainLayer layer;
+    int location = 0;
+
+    public UIItemGrid(ExtraGui guiFor, UIFlatTerrainLayer layerFor) {
+        super(guiFor);
+        layer = layerFor;
+    }
+
+    public void setBlockStateAndOpenParent(ItemStack stack) {
+        // This will work for all vanilla blocks. But for mod blocks we either
+        // need to use dummy world or use itemstacks as layer property and call
+        // "getStateForPlacement(...)" within world generator.
+        this.layer.setBlockState(((ItemBlock) stack.getItem()).block.getStateFromMeta(stack.getMetadata()), stack);
```

Comment:
If you are able to render a block that would be even better. This would also mean that the user will see what the placed block is likely to actually look like for example in case of bed block (but would anyone use that block anyway?)